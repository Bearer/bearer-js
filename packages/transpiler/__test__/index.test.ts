import fs from 'fs'
import Transpiler from '../src'
import path from 'path'
import { idText } from 'typescript'

const fixtures = path.join(__dirname, '__fixtures__')
const preFolder = path.join(fixtures, 'pre')
const postFolder = path.join(fixtures, '../../.build')

const transpiler = new Transpiler(
  __dirname + '/__fixtures__/pre/',
  false,
  '../../../.build/'
)

describe('Transpiler integration test', () => {
  beforeAll(() => {
    process.env.BEARER_SCENARIO_ID = 'SPONGE_BOB'
    fs.readdirSync(postFolder).forEach(file => {
      if (file !== 'tsconfig.json') {
        fs.unlinkSync(path.join(postFolder, file))
      }
    })
    transpiler.run()
  })

  fs.readdirSync(preFolder).forEach(file => {
    describe(file, () => {
      it(`creates a file `, done => {
        expect.assertions(1)
        fs.exists(path.join(postFolder, file), exists => {
          expect(exists).toBe(true)
          done()
        })
      })

      it('match expectation', done => {
        expect.assertions(1)
        fs.readFile(path.join(postFolder, file), 'utf8', (e, postContent) => {
          expect({
            postContent,
            file
          }).toMatchSnapshot()
          done()
        })
      })
    })
  })
})
// test('invoking transpiler', async () => {
//   let transpiler = new Transpiler(SRC_DIRECTORY)
//   await transpiler.run()
//   expect(
//     fs.existsSync(path.join(BUILD_DIRECTORY, 'exportObject.ts'))
//   ).toBeTruthy()
//   expect(
//     fs.existsSync(path.join(BUILD_DIRECTORY, 'classComponent.ts'))
//   ).toBeTruthy()
// })

// test('Adding BEARER_ID prop', async () => {
//   pending('circular calls')
//   let transpiler = new Transpiler(SRC_DIRECTORY)
//   await transpiler.run()
//   const builtFilePath = path.join(BUILD_DIRECTORY, 'classComponent.ts')
// })
