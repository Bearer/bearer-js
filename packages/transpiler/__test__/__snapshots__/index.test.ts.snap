// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpiler integration test FullComponent.tsx match expectation 1`] = `
Object {
  "file": "FullComponent.tsx",
  "postContent": "import { Component, Intent, BearerFetch, IntentType, BearerState, Prop, Watch, State } from '@bearer/core';
@Component({
    tag: 'full-component'
})
export class FullComponent {
    private fetcher: BearerFetch;
    @RetrieveStateIntent()
    retrieve: BearerFetch;
    @BearerState()
    @State()
    attachedPullRequests: Array<any>;
    @BearerState({ statePropName: 'goats' })
    @State()
    ducks: Array<any>;
    private fetchResource: BearerFetch;
    screenRenderer = () => {
        return (<bearer-navigator-screen navigationTitle=\\"Last Screen\\" renderFunc={() => [<h1>Hello Partick</h1>]}/>);
    };
    render() {
        return (<bearer-navigator>
        <bearer-navigator-screen navigationTitle=\\"First Screen\\" renderFunc={() => [<bearer-scrollable fetcher={this.fetcher}/>]}/>
        <bearer-navigator-screen navigationTitle=\\"First Screen\\" renderFunc={() => [<h1>Complex one</h1>,
            <bearer-scrollable fetcher={this.fetcher}/>,
            <span>this text is fine</span>]}/>
        <bearer-navigator-screen navigationTitle={({ data }) => data.name} renderFunc={() => [<bearer-scrollable fetcher={this.fetcher}/>]}/>

        <bearer-navigator-screen renderFunc={({ data, next, prev }) => (<last-screen next={next} complete={({ complete }) => {
            console.log('complete');
            complete();
        }}/>)}/>
        {this.screenRenderer()}
      </bearer-navigator>);
    }
    constructor() { Intent('ListRepositories')(this, \\"fetcher\\"); Intent('getPullRequest', IntentType.GetResource)(this, \\"fetchResource\\"); }
    get SCENARIO_ID() { return \\"SPONGE_BOB\\"; }
    @Prop()
    BEARER_ID: string;
    @Prop({ \\"context\\": \\"bearer\\" })
    bearerContext: any;
    @Prop()
    setupId: string;
    componentDidLoad() { if (this.setupId) {
        this.bearerContext.setupId = this.setupId;
    } }
    @Watch(\\"attachedPullRequests\\")
    _notifyBearerStateHandler_attachedPullRequests(newValue) { this.bearerContext.update(\\"attachedPullRequests\\", newValue); }
    @Watch(\\"ducks\\")
    _notifyBearerStateHandler_ducks(newValue) { this.bearerContext.update('goats', newValue); }
    componentWillLoad() { this.bearerContext.subscribe(this); }
    componentDidUnload() { this.bearerContext.unsubscribe(this); }
    bearerUpdateFromState = state => { this.attachedPullRequests = state[\\"attachedPullRequests\\"]; this.ducks = state['goats']; };
    @Prop()
    referenceId: string;
}
",
}
`;

exports[`Transpiler integration test RootComponent.tsx match expectation 1`] = `
Object {
  "file": "RootComponent.tsx",
  "postContent": "import { Component, Intent, BearerFetch, IntentType, BearerState, Prop, Watch, State } from '@bearer/core';
@Component({
    tag: \\"attach-pull-request-action\\",
    styleUrl: \\"AttachPullRequest.css\\",
    shadow: true
})
export class AttachPullRequestAction {
    private fetcher: BearerFetch;
    @RetrieveStateIntent()
    retrieve: BearerFetch;
    @BearerState()
    @State()
    attachedPullRequests: Array<any>;
    @BearerState({ statePropName: 'goats' })
    @State()
    ducks: Array<any>;
    private fetchResource: BearerFetch;
    render() {
        return <bearer-scrollable fetcher={this.fetcher}/>;
    }
    constructor() { Intent('ListRepositories')(this, \\"fetcher\\"); Intent('getPullRequest', IntentType.GetResource)(this, \\"fetchResource\\"); }
    get SCENARIO_ID() { return \\"SPONGE_BOB\\"; }
    @Prop()
    BEARER_ID: string;
    @Prop({ \\"context\\": \\"bearer\\" })
    bearerContext: any;
    @Prop()
    setupId: string;
    componentDidLoad() { if (this.setupId) {
        this.bearerContext.setupId = this.setupId;
    } }
    @Watch(\\"attachedPullRequests\\")
    _notifyBearerStateHandler_attachedPullRequests(newValue) { this.bearerContext.update(\\"attachedPullRequests\\", newValue); }
    @Watch(\\"ducks\\")
    _notifyBearerStateHandler_ducks(newValue) { this.bearerContext.update('goats', newValue); }
    componentWillLoad() { this.bearerContext.subscribe(this); }
    componentDidUnload() { this.bearerContext.unsubscribe(this); }
    bearerUpdateFromState = state => { this.attachedPullRequests = state[\\"attachedPullRequests\\"]; this.ducks = state['goats']; };
    @Prop()
    referenceId: string;
}
",
}
`;

exports[`Transpiler integration test exportObject.ts match expectation 1`] = `
Object {
  "file": "exportObject.ts",
  "postContent": "class SpongeBobHelper {
    private name: string;
    constructor(name: string) {
        this.name = name;
    }
}
export class PatrickHelper {
    constructor() { }
}
export default {
    spongeHelper: SpongeBobHelper,
    patrickHelper: PatrickHelper
};
",
}
`;

exports[`Transpiler integration test invalidRootComponent.tsx match expectation 1`] = `
Object {
  "file": "invalidRootComponent.tsx",
  "postContent": "import { Component, Intent, BearerFetch, IntentType, BearerState, Prop, Watch, State } from '@bearer/core';
@Component({
    tag: \\"attach-pull-request-action\\",
    styleUrl: \\"AttachPullRequest.css\\",
    shadow: true
})
export class AttachPullRequestAction {
    private fetcher: BearerFetch;
    @RetrieveStateIntent()
    retrieve: BearerFetch;
    @BearerState()
    @State()
    attachedPullRequests: Array<any>;
    @BearerState({ statePropName: 'goats' })
    @State()
    ducks: Array<any>;
    private fetchResource: BearerFetch;
    render() {
        return <bearer-scrollable fetcher={this.fetcher}/>;
    }
    constructor() { Intent('ListRepositories')(this, \\"fetcher\\"); Intent('getPullRequest', IntentType.GetResource)(this, \\"fetchResource\\"); }
    get SCENARIO_ID() { return \\"SPONGE_BOB\\"; }
    @Prop()
    BEARER_ID: string;
    @Prop({ \\"context\\": \\"bearer\\" })
    bearerContext: any;
    @Prop()
    setupId: string;
    componentDidLoad() { if (this.setupId) {
        this.bearerContext.setupId = this.setupId;
    } }
    @Watch(\\"attachedPullRequests\\")
    _notifyBearerStateHandler_attachedPullRequests(newValue) { this.bearerContext.update(\\"attachedPullRequests\\", newValue); }
    @Watch(\\"ducks\\")
    _notifyBearerStateHandler_ducks(newValue) { this.bearerContext.update('goats', newValue); }
    componentWillLoad() { this.bearerContext.subscribe(this); }
    componentDidUnload() { this.bearerContext.unsubscribe(this); }
    bearerUpdateFromState = state => { this.attachedPullRequests = state[\\"attachedPullRequests\\"]; this.ducks = state['goats']; };
    @Prop()
    referenceId: string;
}
",
}
`;

exports[`Transpiler integration test overrides-decorator.tsx match expectation 1`] = `
Object {
  "file": "overrides-decorator.tsx",
  "postContent": "import { Component, Intent, BearerFetch, IntentType, Prop } from '@bearer/core';
@Component({
    tag: 'overrides-decorator-full'
})
class OverridesDecorarotFull {
    private fetcher: BearerFetch;
    private memberFetcher: BearerFetch;
    constructor() { Intent('collectionIntent')(this, \\"fetcher\\"); Intent('collectionIntent', IntentType.GetResource)(this, \\"memberFetcher\\"); }
    componentWillLoad() {
        console.log('componentWillLoad');
    }
    componentDidUnload() {
        console.log('componentDidUnload');
    }
    get SCENARIO_ID() { return \\"SPONGE_BOB\\"; }
    @Prop()
    BEARER_ID: string;
    @Prop({ \\"context\\": \\"bearer\\" })
    bearerContext: any;
    @Prop()
    setupId: string;
    componentDidLoad() { if (this.setupId) {
        this.bearerContext.setupId = this.setupId;
    } console.log('componentDidLoad'); }
}
",
}
`;

exports[`Transpiler integration test overrides-decorator-less.1.tsx match expectation 1`] = `
Object {
  "file": "overrides-decorator-less.1.tsx",
  "postContent": "import { Component, Prop } from '@bearer/core';
@Component({
    tag: 'overrides-decorator-less'
})
class OverridesDecorarotLess {
    constructor() { }
    componentWillLoad() {
        console.log('componentWillLoad');
    }
    componentDidUnload() {
        console.log('componentDidUnload');
    }
    get SCENARIO_ID() { return \\"SPONGE_BOB\\"; }
    @Prop()
    BEARER_ID: string;
    @Prop({ \\"context\\": \\"bearer\\" })
    bearerContext: any;
    @Prop()
    setupId: string;
    componentDidLoad() { if (this.setupId) {
        this.bearerContext.setupId = this.setupId;
    } console.log('componentDidLoad'); }
}
",
}
`;
